situp-plugins/lmdb-processor-state/src/main/java/com/amazon/situp/plugins/processor/state/LmdbProcessorState.java

```diff
@@ -145,15 +145,30 @@ public long size() {
         }
     }
 
+    private long[] getRange(int segments, int index) {
```

Comment:
you should return some sort of pair, I know Java is not the cleanest for this but the array is not intuitive

Reply:
^ +1, maybe something like `KeyRange` object

Reply:
Good point, ill make a KeyRange object. Pair requires javafx, which i'd rather not pull in if i dont have to